src/steps/generic.rs

```diff
@@ -1318,20 +1318,21 @@ pub fn run_uv(ctx: &ExecutionContext) -> Result<()> {
         .execute(&uv_exec)
         .arg("--version")
         .output_checked_utf8()?;
-    // example output: "uv 0.5.11 (c4d0caaee 2024-12-19)\n"
+    // Multiple possible output formats are possible. For example:
+    //  "uv 0.5.11 (c4d0caaee 2024-12-19)\n"
+    //  "uv 0.6.14\n"
     let uv_version_output_stdout = uv_version_output.stdout;
 
     let version_str = {
-        // trim the starting "uv" and " " (whitespace)
+        // Trim the starting "uv" and " " (whitespace)
         let start_trimmed = uv_version_output_stdout
             .trim_start_matches("uv")
             .trim_start_matches(' ');
-        // remove the tailing part " (c4d0caaee 2024-12-19)\n"
-        let first_whitespace_index = start_trimmed
-            .find(' ')
-            .expect("the output of `uv --version` changed, please file an issue to Topgrade");
-        // this should be our version str "0.5.11"
-        &start_trimmed[..first_whitespace_index]
+        // Remove the tailing part " (c4d0caaee 2024-12-19)\n", if it's there
+        match start_trimmed.find(' ') {
+            None => start_trimmed.trim_end_matches('\n'), // Otherwise, just strip the newline
+            Some(i) => &start_trimmed[..i],               // This should be our version str "0.5.11"
+        }
```

Comment:
```suggestion
        // Remove the tailing part " (c4d0caaee 2024-12-19)\n", if it's there
        match start_trimmed.find(' ') {
            None => start_trimmed.trim_end_matches('\n'), // Otherwise, just strip the newline
            Some(i) => &start_trimmed[..i],
        }
        
        // After trimming, it should be a string in 2 possible formats, both can be handled by `Version::parse()`
        //
        // 1. "0.5.11"
        // 2. "0.5.11+1"
```